Doc for Autoupgrade a module from local source (#1705)
PrestaShop/docs: basics/keeping-up-to-date/upgrade-module/upgrade-module-internal-behavior.md

- Classes interacting with PrestaShop core
- Tasks (Can be considered as controllers for upgrade, rollback etc.)

## Upgrade modules from local source

During the upgrade process, we check for new module versions in the PrestaShop Marketplace. If there is one, we download it and then upgrade it.

> 

Reviewer: ```suggestion
```

- Classes interacting with PrestaShop core
- Tasks (Can be considered as controllers for upgrade, rollback etc.)

## Upgrade modules from local source

> During the autoupgrade process, in the "modules upgrade" step, we check for each module if there is a new version available in the PrestaShop Marketplace

Reviewer: ```suggestion
During the upgrade process, we check for new module versions in the PrestaShop Marketplace. If there is one, we download it and then upgrade it.
```

Reviewer: Is it only about the native modules? How does it work if you have a paid module and you are not logged in to your Addons account?

Author: I don't change the modules upgrade process

I guess if we have a paid module and the user is not logged to addons/account, the new version download fails

Author: If there is a zip given, I don't check the Addons connection

> If there is one, we download it and then upgrade.

Reviewer: ```suggestion
```

> That means, for developers, to test the autoupgrade process with your module, you must release it and push it to the Marketplace before.
> 
> To ease the tests, you can put your module ZIP in `/ADMIN_DIR/autoupgrade/modules/MODULE_NAME.zip` and then use your local version to upgrade.

Reviewer: ```suggestion
Developers must release and push their module to the Marketplace in order to test the upgrade process, which is inconvenient. In order to simplify the testing process, you can place the ZIP file of your module in `/ADMIN_DIR/autoupgrade/modules/MODULE_NAME.zip`. This will allow you to use your local version for upgrading purposes.
```

> If there is any failure in using your local archive, the new version will be pulled from the Marketplace.

Reviewer: ```suggestion
In case your local archive with a module fails to work, the latest version will be downloaded from the Marketplace.
```
